app/src/lib/stores/app-store.ts

```diff
@@ -5018,29 +5027,44 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
-  /**
-   *  Note: **drops the existing desktop stash** on that branch (if there is one)
-   */
-  private async createStash(repository: Repository, branchName: string) {
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _createStash(
```

Comment:
I don't think the callback pattern is serving us here. `_createStash` is only called in two places, and in each place one of the callbacks is omitted. This means we're only sharing this line between the two codepaths:

```js
await createDesktopStashEntry(repository, branchName)
```

I think splitting this into two methods would be much less complex and would let us avoid inlining callback methods in `error-handlers.ts` and `_checkoutBranch`.

Reply:
Addressed in f5edd65.